README.md

```diff
@@ -1,64 +1,199 @@
-provided by the bot. You will only need to do this once across all repos using our CLA.
-
-This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
-For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or
-contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+---
+services: active-directory
+platforms: java
+author: ramya25
+level: 200
+client: java console app
+service: Microsoft Graph
+endpoint: Microsoft identity platform
+page_type: sample
+languages:
+  - java  
+products:
+  - azure
+  - azure-active-directory  
+  - office-ms-graph
+description: "This sample demonstrates a java console application calling The Microsoft Graph"
+---
+# Invoking an API protected by Azure AD from a text-only device
+
+## About this sample
+
+### Overview
+
+This sample shows how to use the OAuth 2.0 [device code flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-device-code)to sign in users to input-constrained devices such as a smart TV, IoT device, or printer. It leverages [MSAL for Java (MSAL4J)](https://github.com/AzureAD/microsoft-authentication-library-for-java) from an app that **does not have the capability of offering an interactive authentication experience**. It enables these apps to:
+
+- Authenticate a user
+- Acquire an [Access token](https://docs.microsoft.com/azure/active-directory/develop/access-tokens) for a web API (in this case, the [Microsoft Graph](https://graph.microsoft.com))
+
+To emulate a an app on a device that is not capable of showing a UX, the sample is packaged as a Java console application.
+The application signs-in users with Azure Active Directory (Azure AD), using the Microsoft Authentication Library for Java (MSAL4J) to obtain a JWT access token through the OAuth 2.0 protocol. The access token is then used to call the Microsoft Graph API to obtain information about the user who signed-in. The sample is structured as such so that you can call your own API.
+
+![Topology](./ReadmeFiles/Topology.png)
+
+### Scenario
+
+The application obtains tokens through a two step process especially designed for devices and operating systems that cannot display any UX. Examples of such applications are applications running on iOT, or Command-Line tools (CLI). The idea is that:
+
+1. Whenever a user is required to authenticate, the command-line app provides a code and asks the user to use another device (such as an internet-connected smartphone) to navigate to [https://microsoft.com/devicelogin](https://microsoft.com/devicelogin), where the user will be prompted to enter the code provided earlier. Oce the user enters the code, the web page will lead the user through a normal authentication experience, including presenting any consent prompts and take the user through a multi-factor authentication experience if necessary.
```

Comment:
Once*

"...and taking* the user through a multi-factor..."

Reply:
corrected